From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aq17 <aqiu@pulumi.com>
Date: Thu, 25 May 2023 10:33:39 -0700
Subject: [PATCH] Expose provider via shim

Add shim module containing a public method which constructs a new instance of the internal provider class so we're able to access the provider class directly in order to bridge it.

diff --git a/shim/shim.go b/shim/shim.go
new file mode 100644
index 0000000000..e5291c4749
--- /dev/null
+++ b/shim/shim.go
@@ -0,0 +1,14 @@
+package shim
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/provider"
+)
+
+func NewProvider() *schema.Provider {
+	return provider.AzureProvider()
+}
+
+// expose client type
+type AzureClient = clients.Client
